### PR TITLE
util-linux: update to 2.38

### PR DESCRIPTION
**Pull request by neheb**
Use the newly merged meson system for faster compilation.

Add various fixes to it as it seems cross compilation was never tested.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

https://github.com/util-linux/util-linux/pull/1672

### CODE REVIEW

**Comment by neheb** [comment]
Before:

```
time make package/util-linux/{clean,compile} -j 20
Executed in   44.09 secs    fish           external
   usr time   99.13 secs    0.00 micros   99.13 secs
   sys time   14.99 secs  382.00 micros   14.99 secs
```

After:

```
time make package/util-linux/{clean,compile} -j 20
Executed in   27.53 secs    fish           external
   usr time   65.31 secs  359.00 micros   65.31 secs
   sys time    8.82 secs   87.00 micros    8.82 secs
```

File: package/utils/util-linux/Makefile [hunk 1]
```diff
@@ -552,44 +548,69 @@ define Package/wipefs/description
...
-	$(SED) 's,/usr/lib,$$$${exec_prefix}/lib,g' $(1)/usr/lib/pkgconfig/fdisk.pc
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/pkgconfig/mount.pc $(1)/usr/lib/pkgconfig
-	$(SED) 's,/usr/include,$$$${prefix}/include,g' $(1)/usr/lib/pkgconfig/mount.pc
-	$(SED) 's,/usr/lib,$$$${exec_prefix}/lib,g' $(1)/usr/lib/pkgconfig/mount.pc
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/pkgconfig/smartcols.pc $(1)/usr/lib/pkgconfig
-	$(SED) 's,/usr/include,$$$${prefix}/include,g' $(1)/usr/lib/pkgconfig/smartcols.pc
-	$(SED) 's,/usr/lib,$$$${exec_prefix}/lib,g' $(1)/usr/lib/pkgconfig/smartcols.pc
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/pkgconfig/uuid.pc $(1)/usr/lib/pkgconfig
-	$(SED) 's,/usr/include,$$$${prefix}/include,g' $(1)/usr/lib/pkgconfig/uuid.pc
-	$(SED) 's,/usr/lib,$$$${exec_prefix}/lib,g' $(1)/usr/lib/pkgconfig/uuid.pc
```
Comment (hauke) [inline package/utils/util-linux/Makefile:592]: Are the.pc files still correct?

File: package/utils/util-linux/Makefile [hunk 2]
```diff
@@ -8,12 +8,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=util-linux
-PKG_VERSION:=2.37.3
+PKG_VERSION:=2.38
```
Comment (hauke) [inline package/utils/util-linux/Makefile:11]: I would prefer if you split this into to commits, one doing the upgrade and an other doing the move to meson.

File: package/utils/util-linux/Makefile (hunk 1 above)
Comment (neheb) [reply package/utils/util-linux/Makefile:592]: Yes

**Comment by breakings** [comment]
dmesg 2.38 by default not use` -L `show color output.

File: package/utils/util-linux/Makefile (hunk 2 above)
Comment (neheb) [reply package/utils/util-linux/Makefile:11]: Done.

**Comment by nasbdh9** [comment]
```
The Meson build system
Version: 0.61.4
Source dir: /home/nasbdh9/openwrt/build_dir/target-x86_64_musl/util-linux-2.38
Build dir: /home/nasbdh9/openwrt/build_dir/target-x86_64_musl/util-linux-2.38/openwrt-build
Build type: cross build
Project name: util-linux
Project version: 2.37
C compiler for the host machine: x86_64-openwrt-linux-musl-gcc (gcc 12.1.0 "x86_64-openwrt-linux-musl-gcc (OpenWrt GCC 12.1.0 r19662-1ab2ce32a9) 12.1.0")
C linker for the host machine: x86_64-openwrt-linux-musl-gcc ld.bfd 2.38
C compiler for the build machine: gcc (gcc 9.4.0 "gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0")
C linker for the build machine: gcc ld.bfd 2.34
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: x86_64
Host machine cpu: generic
Target machine cpu family: x86_64
Target machine cpu: generic
Dependency bash-completion skipped: feature build-bash-completion disabled
WARNING: You should add the boolean check kwarg to the run_command call.
         It currently defaults to false,
         but it will default to true in future releases of meson.
         See also: https://github.com/mesonbuild/meson/issues/9300
Checking if "wchar_t support" compiles: YES
Library m found: YES
Dependency tinfo skipped: feature tinfo disabled
Found pkg-config: /home/nasbdh9/openwrt/staging_dir/host/bin/pkg-config (1.8.0)
Run-time dependency ncursesw found: YES 6.3.20211021
Dependency slang skipped: feature slang disabled
Checking for function "use_default_colors" with dependency ncursesw: YES
Checking for function "resizeterm" with dependency ncursesw: YES
Run-time dependency zlib found: YES 1.2.12
Dependency readline skipped: feature readline disabled
Dependency libuser skipped: feature libuser disabled
Library util skipped: feature libutil disabled
Library utempter skipped: feature libutempter disabled
Dependency systemd skipped: feature systemd disabled
Dependency libsystemd skipped: feature systemd disabled
Dependency libudev skipped: feature systemd disabled
Library crypt found: YES
Library pam found: NO
Dependency libcryptsetup skipped: feature cryptsetup disabled
Checking for function "crypt_activate_by_signed_key" with dependency libcryptsetup: NO
Found CMake: /home/nasbdh9/openwrt/staging_dir/host/bin/cmake (3.22.3)
Run-time dependency libcap-ng found: NO (tried pkgconfig and cmake)

../../../build_dir/target-x86_64_musl/util-linux-2.38/meson.build:336:0: ERROR: Dependency "libcap-ng" not found, tried pkgconfig and cmake

A full log can be found at /home/nasbdh9/openwrt/build_dir/target-x86_64_musl/util-linux-2.38/openwrt-build/meson-logs/meson-log.txt
make[3]: *** [Makefile:878: /home/nasbdh9/openwrt/build_dir/target-x86_64_musl/util-linux-2.38/.configured_68b329da9893e34099c7d8ad5cb9c940] Error 1
```

libcap-ng is missing?

**Comment by neheb** [comment]
Hmm I fixed this upstream: https://github.com/util-linux/util-linux/commit/cd23a4336f49ba6a12ade557a09589f2a7c966f4

Thought I backported it here.

**Comment by neheb** [comment]
@nasbdh9 fixed.

**Comment by chunkeey** [comment]
Merged. Thanks.

**Pull request closed without merging**